src/__tests__/apply.test.js

```diff
@@ -0,0 +1,22 @@
+/* eslint-env node, jest */
+const { apply } = require('..')
+
+describe('apply', function () {
+  let testFn = jest.fn()
```

Comment:
Mocked functions should not be shared between multiple units. You should move this line to each `it` block instead.

Reply:
@conartist6 You still have this to be resolved. Mocked functions contain side-effects, which should **never** be shared by multiple independant tests.

Reply:
@KSXGitHub this should take care of it:
```js
  beforeEach(function () {
    jest.resetAllMocks()
  })
```
I am still quite new to jest, so I don't know if this is really an issue.

Reply:
Resetting all mocks is not only unnecessary, it would even cause logic error.

Resetting just `testFn` alone is okay, but I still think side-effects should not be shared.

Reply:
> it would even cause logic error.

how so?

Reply:
> > it would even cause logic error.
> 
> how so?

It won't right now, but it will when we add more mocks to the test.

Reply:
Even if it never does, I still prefer a more straightforward code. Imagine reading through all the `beforeEach`/`beforeAll`/`afterEach`/`afterAll` and try to reason with it! I personally never use unnecessary `beforeEach` unless I have to make side-effects (such as modifying filesystem, etc.)

Reply:
I see your point about making the code more straightforward. From the context I don't see how this can really make any difference. But it is such a small change and I am ok doing it.